src/repository.js

```diff
@@ -105,7 +105,7 @@ export class Repository {
     }
 
     return findQuery
-      .then(x => this.populateEntities(x))
+      .then(x => x ? this.populateEntities(x) : Promise.reject(new Error('No data')))
```

Comment:
i always get that confused. but right i think rejection with non-error was what one shouldn't do